main/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 add_executable(main)
 target_sources(main PRIVATE main.cpp)
-target_link_libraries(main PPN-microbench nlohmann_json::nlohmann_json)
+target_link_libraries(main PPN-microbench nlohmann_json::nlohmann_json )
```

Comment:
Formatting change but why the final space?
Also, consider adding `PRIVATE` after `main` to indicate that these dependencies should not be forwarded to anything that depends on `main`.